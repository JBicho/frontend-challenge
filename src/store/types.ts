export interface ITranscription {
  id: number;
  voice: string;
  text: string;
}

export interface ITranscriptionState {
  transcriptionList: ITranscription[];
}

export interface IRootState {
  transcriptions: {
    list: ITranscription[];
  };
}
