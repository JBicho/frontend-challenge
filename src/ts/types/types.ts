export interface ITranscription {
  id: number;
  voice: string;
  text: string;
}
