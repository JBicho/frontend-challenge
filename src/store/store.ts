import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import { IRootState } from './types';
import { transcription } from './modules/transcriptions';

Vue.use(Vuex);

const store: StoreOptions<IRootState> = {
  state: {
    transcriptions: {
      list: []
    }
  },
  modules: {
    transcription
  }
};

export default new Vuex.Store<IRootState>(store);
