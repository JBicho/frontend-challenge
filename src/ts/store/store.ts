import { createBrowserHistory } from "history";
import { routerMiddleware } from "react-router-redux";
import { applyMiddleware, combineReducers, compose, createStore } from "redux";
import createSagaMiddleware from "redux-saga";
import { all } from "redux-saga/effects";
import { transcriptionsReducer } from "./transcriptions/reducer";
import { transcriptionsSaga } from "./transcriptions/saga";
import { ITranscriptionsState } from "./transcriptions/state";

export interface IAppState {
  transcriptionsState: ITranscriptionsState;
}

const sagaMiddleware = createSagaMiddleware();

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const browserHistory = createBrowserHistory();

export const store = createStore(
  combineReducers<IAppState>({
    transcriptionsState: transcriptionsReducer
  }),
  composeEnhancers(
    applyMiddleware(sagaMiddleware, routerMiddleware(browserHistory))
  )
);

sagaMiddleware.run(function*() {
  yield all([transcriptionsSaga()]);
});
