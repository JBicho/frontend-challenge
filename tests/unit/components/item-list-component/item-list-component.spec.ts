import Vue from 'vue';
import { shallowMount, createLocalVue, mount } from '@vue/test-utils';
import ItemsList from '@/components/items-list-component/items-list-component.vue';
import Vuex from 'vuex';
import { ITranscription } from '@/store/types';

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuex);

describe('Item list component tests', () => {
  let wrapper: any;
  let store: any;
  let actions: any;
  let state: any;
  let transcriptionList: ITranscription[];

  beforeEach(() => {
    transcriptionList = [
      {
        id: 1,
        voice: 'Test Voice',
        text: 'Test description'
      },
      {
        id: 2,
        voice: 'Test Voice 2',
        text: 'Test description 2'
      }
    ];

    actions = {
      deleteTranscription: jest.fn(() => true)
    };

    state = {
      transcriptionList: [],
      errors: []
    };

    store = new Vuex.Store({
      state: {},
      actions,
      modules: {
        transcription: {
          namespaced: true,
          state,
          actions
        }
      }
    });
  });

  test('Snapshot test', () => {
    const Constructor = Vue.extend(ItemsList);
    const vm = new Constructor().$mount();
    expect(vm.$el).toMatchSnapshot();
  });

  test('It should render a list of items passed as props', () => {
    wrapper = mount(ItemsList, {
      store,
      localVue,
      propsData: { transcriptionList }
    });

    expect(wrapper.findAll('li').length).toBe(2);
  });

  test('It should not render a list of items when an empty array is passed in props', () => {
    const emptyList: any[] = [];

    wrapper = mount(ItemsList, {
      store,
      localVue,
      propsData: { emptyList }
    });

    expect(wrapper.findAll('li').length).toBe(0);
  });

  test('Should pass delete action to state', () => {
    wrapper = mount(ItemsList, {
      store,
      localVue,
      propsData: { transcriptionList }
    });

    wrapper.find('.delete-button').trigger('click');

    expect(actions.deleteTranscription).toHaveBeenCalled();
  });
});
