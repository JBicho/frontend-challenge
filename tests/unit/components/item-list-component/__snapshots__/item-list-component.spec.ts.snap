// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item list component tests Snapshot test 1`] = `
<ul
  class="item-list"
/>
`;
